docs/data/base/components/badge/badge.md

```diff
@@ -22,28 +22,31 @@ import BadgeUnstyled from '@mui/base/BadgeUnstyled';
 
 ## Badge visibility
 
-The visibility of badges can be controlled using the `invisible` prop.
-If a badge is invisible, it has the `MuiBadge-invisible` class.
-It is up to the developer to provide styles that actually hide the badge.
+You can control the visibility of a `BadgeUnstyled` by using the `invisible` prop.
+Setting a badge to `invisible` does not actually hide it—instead, this prop adds the `MuiBadge-invisible` class to the badge, which you can target with styles to hide however you prefer:
 
 {{"demo": "BadgeVisibility.js"}}
 
-The badge hides automatically when `badgeContent` is zero. You can override this with the `showZero` prop.
+## Numerical badges
 
-{{"demo": "ShowZeroBadge.js"}}
+The following props are useful when `badgeContent` is a number.
+
+### The showZero prop
 
-## Maximum value
+By default, badges automatically hide when `badgeContent={0}`. You can override this behavior with the `showZero` prop:
+
+{{"demo": "ShowZeroBadge.js"}}
 
-You can use the `max` prop to cap the value of the badge content.
-It is set to 99 by default.
+### The max prop
 
-Note that `badgeContent` should be a number (or convertible to a number) for this to work.
+You can use the `max` prop to set a maximum value for `badgeContent`.
+The default is 99.
 
 {{"demo": "BadgeMax.js"}}
 
 ## Accessibility
 
 You can't rely on the content of the badge to be announced correctly.
```

Comment:
Can we elaborate on this? Why is it not reliable? What happens if you omit the `aria-label`?

Reply:
If the aria-label is omitted, a screen reader would read the contents of the badge verbatim ("ninety-nine plus button" in this case), which may not give a user enough information to figure out what the focused component is for.

Reply:
That makes sense. Thanks for the clarification!